odoo/addons/base/models/res_users.py

```diff
@@ -665,6 +665,10 @@ def context_get(self):
 
         return frozendict(context)
 
+    @tools.ormcache('self._uid')
+    def _get_company_ids(self):
+        return self.company_ids.ids
```

Comment:
- the cache key is wrong? (considers current env user rather than target)
- where is it invalidated?
- it's brittle, how would business code know to use this?

If the point is to save some queries for the website user, why not make a cache that is specific to the website user?

Reply:
Currently this is used for all calls to `env.company` and `env.companies` in addition to a call to website.
I had misread, indeed the key is not in the fields of `_get_invalidation_fields`.

Reply:
@odony see the last commit for your review.